tests/dummy/app/templates/docs/faq/routing-and-asset-loading.md

```diff
@@ -0,0 +1,101 @@
+# Routing and Asset Loading
+
+This is a bit of a deep dive into how we make Ember's routing and asset loading work properly when running in an Electron application, along with a discussion of some possible alternatives. This all stems from the fact that in an Electron application we load the Ember app using a `file:` URL rather than an `http:` URL.
+
+## Routing
+
+Let's say our app uses the default `history` (or `auto`) location type, and has the following routes:
+
+```javascript
+Router.map(function() {
+  this.route('my-route');
+});
+```
+
+and we deploy it to `http://my-app.com`. This means that the URL `http://my-app.com/` corresponds to the `index` route, and `http://my-app.com/my-route` corresponds to the `my-route` route. The way this is achieved is that whatever web server is serving the Ember app is typically configured to return the Ember app's `index.html` for any requests to `http://my-app.com/<subpath>`, and `index.html` loads all the Javascript for the application, including Ember itself. Then Ember's routing code looks at the URL path and activates the corresponding route. So the web server combined with Ember's `history` location/routing code creates an abstraction layer that interprets logical locations, in the form of the URL path, resolves them all to the same `index.html` file, but in a way that exposes that logical location to Ember so it can activate the correct route.
+
+When running an Electron app, we serve the files out of the filesystem using a `file:` URL, which doesn't give us the flexibility of a web server -- the URL path cannot be a logical location, but must be a filesystem path that points directly to `index.html`. For example, when running a development build, the path might be `file:///Users/me/projects/my-app/electron-app/ember-dist/index.html`. We can't use the URL path to specify the Ember route because telling the browser to load `file:///Users/me/projects/my-app/electron-app/ember-dist/index.html/my-route` will not work -- it will look for a file named `my-route` inside a directory named `index.html`, and won't find it.
+
+So we have to go back to an older method of specifying logical locations, which is to use the URL hash. Fortunately, Ember supports this via the [HashLocation](https://api.emberjs.com/ember/release/classes/HashLocation) since that was the only option in older browsers. Since URL hashes don't affect how `file:` URLs are resolved to paths, we can tell the browser to load `file:///Users/me/projects/my-app/electron-app/ember-dist/index.html#my-route`, and it will load the `index.html` and then Ember can resolve the route using the path contained in the URL hash.
```

Comment:
```suggestion
So we have to go back to an older method of specifying logical locations, which is to use the URL hash. Fortunately, Ember supports this via the [HashLocation](https://api.emberjs.com/ember/release/classes/HashLocation) since that was the only option in older browsers. Since URL hashes don't affect how `file:` URLs are resolved to paths, we can tell the browser to load `file:///Users/me/projects/my-app/electron-app/ember-dist/index.html#/my-route`, and it will load the `index.html` and then Ember can resolve the route using the path contained in the URL hash.
```

Hash routes need to be `index.html#/route-name`